src/main/webapp/src/App.js

```diff
@@ -10,35 +10,60 @@ import Phones from "./components/Phones";
 import MainPage from "./components/MainPage";
 import PhoneDetails from "./components/PhoneDetails";
 import NotFoundPage from './components/NotFoundPage';
+import Basket from './components/Basket';
+
 import {Redirect} from "react-router";
+import axios from "axios";
 
 const {Header, Footer, Content} = Layout;
 
+class App extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            basket: null,
+            numberItems: 0,
```

Comment:
why do you add this state variable. use basket.items.length if basket is not null and 0 if basket is null

Reply:
+